### PR TITLE
[Merged by Bors] - chore: do not directly import `Tactic.Cases`

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
This entails replacing not only `induction'` by its unprimed variant, but also `cases'` by `cases/rcases/obtain`. This reduces the number of imports for some files.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Size.lean [hunk 1]
```diff
@@ -57,8 +57,9 @@ end
 @[simp]
 theorem size_shiftLeft' {b m n} (h : shiftLeft' b m n ≠ 0) :
     size (shiftLeft' b m n) = size m + n := by
-  induction' n with n IH
-  · simp [shiftLeft']
+  induction n with
+  | zero => simp [shiftLeft']
+  | succ n IH =>
```
Comment (semorrison) [inline Mathlib/Data/Nat/Size.lean:62]: Indent the remaining proof?

File: Mathlib/Order/WithBot.lean [hunk 2]
```diff
@@ -338,16 +338,18 @@ instance preorder [Preorder α] : Preorder (WithBot α) where
     let ⟨c, hc, bc⟩ := h₂ b hb
     ⟨c, hc, le_trans ab bc⟩
 
-instance partialOrder [PartialOrder α] : PartialOrder (WithBot α) :=
-  { WithBot.preorder with
```
Comment (semorrison) [inline Mathlib/Order/WithBot.lean:342]: For simplicity, best not to make random change like the `__` change here. (Potentially subtle consequences, and orthogonal to everything else, so make an easy to review PR not quite as easy.)

**Comment by semorrison** [comment]
bors d+

**Comment by Parcly-Taxel** [comment]
bors merge

**Pull request closed without merging**